.github/workflows/e2e-hec-exporter-tests.yaml

```diff
@@ -0,0 +1,103 @@
+name: e2e-hec-exporter-tests
+
+on: 
+  pull_request
+
+jobs:
+  e2e-test:
+    name: HEC Exporter e2e tests
+    runs-on: ubuntu-latest
+
+    env:
+      CI_SPLUNK_PORT: 8089
+      CI_SPLUNK_USERNAME: admin
+      CI_SPLUNK_HEC_TOKEN: 00000000-0000-0000-0000-000000000000
+      CI_SPLUNK_HEC_TOKEN_DISABLED: f035f5d0-c141-4d83-9f57-00073b992c74
+      CI_SPLUNK_PASSWORD: helloworld
+      CI_INDEX_EVENTS: sck-otel
+      CI_INDEX_TIMESTAMP_EVENTS: timestamp
+      CI_INDEX_EVENTS_FILE_LOGS: file_logs
+      CI_INDEX_METRICS_HOST_METRICS_EVENTS: metrics
+      CI_INDEX_METRICS_HEC_RECEIVER_EVENTS: sck-metrics
+      CONTAINER_RUNTIME: docker
+      KUBERNETES_VERSION: v1.26.2
+      MINIKUBE_VERSION: v1.30.1
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Setup Minikube
```

Comment:
Why do we need to spin up kubernetes here? Can we implement this as an integration test and run the Splunk instance in a container using github.com/testcontainers/testcontainers-go? Please check how other integration tests are written for similar use cases.

We'd like to have a consistent approach for testing components rather than having a custom Github workflow for each one. Could you please take a look into this?

Reply:
I have copied the this part from SCK-OTEL pipeline. 
I can change it.
After spinning Splunk we need to configure it: set HEC tokens, add indexes) -> currently this is done using curl 
Can we spin Splunk inside docker container instead of using testcontainers-go? 
I think that it will be easier to use docker container here.

Reply:
> Can we spin Splunk inside docker container instead of using testcontainers-go?

testcontainers-go is actually a handy wrapper to run docker containers. Did you run into any issues with using it?

Reply:
Actually I have created a working example which spins up Splunk, I have tested it locally and on Nova EC2 instance - it is working. But I am not sure if it fits use case here. The HEC exporter e2e pipeline works as follows:
1. Spin Splunk instance
2. Configure it (indexes, HEC)
3. Get Splunk IP and spin OTEL contrib collector 
4. Run tests
We need running Splunk instance before starting collector (step no.3 ) - collector will not start if it will not be able to establish connection to Splunk

I am not sure if I understand correctly, what should I do here:
1. rewrite step no.1 into golang and do not kill Splunk container and then proceed with steps no. 2 and 3 as they are now
or maybe
2. rewrite everything into golang step 1, 2 and 3 - it will take more time

Reply:
My suggestion is to write all the steps in golang using the go-testcontainers to spin up Splunk in docker

Reply:
Test have been redesigned.
New redesigned tests which are spinning Splunk using go-testscontainers and are part of integration test suite can be found on new branch: 
https://github.com/pszkamruk-splunk/opentelemetry-collector-contrib/tree/hec-exporter-integration-test-initial-branch

PR to merge tests into main repo:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23637